bundles/org.openhab.binding.ferroamp/src/main/java/org/openhab/binding/ferroamp/internal/FerroampHandler.java

```diff
@@ -0,0 +1,778 @@
+@NonNullByDefault
+public class FerroampHandler extends BaseThingHandler implements MqttMessageSubscriber {
+    private final Logger logger = LoggerFactory.getLogger(FerroampHandler.class);
+
+    private @Nullable FerroampConfiguration ferroampConfig;
+    private @Nullable MqttBrokerConnection ferroampConnection;
+
+    private List<FerroampChannelConfiguration> channelConfigEhub = new ArrayList<>();
+    private List<FerroampChannelConfiguration> channelConfigSsoS0 = new ArrayList<>();
+    private List<FerroampChannelConfiguration> channelConfigSsoS1 = new ArrayList<>();
+    private List<FerroampChannelConfiguration> channelConfigSsoS2 = new ArrayList<>();
+    private List<FerroampChannelConfiguration> channelConfigSsoS3 = new ArrayList<>();
+    private List<FerroampChannelConfiguration> channelConfigEso = new ArrayList<>();
+    private List<FerroampChannelConfiguration> channelConfigEsm = new ArrayList<>();
+
+    int ssoIdSet = 0;
+    boolean establishedConnection = false;
+
+    public FerroampHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        String transId = UUID.randomUUID().toString();
+        String valueConfiguration = command.toString();
+
+        if (FerroampBindingConstants.CHANNEL_REQUESTCHARGE.equals(channelUID.getId())) {
+            String requestCmdJsonCharge = "{\"" + "transId" + "\":\"" + transId
+                    + "\",\"cmd\":{\"name\":\"charge\",\"arg\":\"" + valueConfiguration + "\"}}";
+            sendMQTT(requestCmdJsonCharge);
+        }
+        if (FerroampBindingConstants.CHANNEL_REQUESTDISCHARGE.equals(channelUID.getId())) {
+            String requestCmdJsonDisCharge = "{\"" + "transId" + "\":\"" + transId
+                    + "\",\"cmd\":{\"name\":\"discharge\",\"arg\":\"" + valueConfiguration + "\"}}";
+            sendMQTT(requestCmdJsonDisCharge);
+        }
+        if (FerroampBindingConstants.CHANNEL_AUTO.equals(channelUID.getId())) {
+            String requestCmdJsonAuto = "{\"" + "transId" + "\":\"" + transId + "\",\"cmd\":{\"name\":\"auto\"}}";
+            sendMQTT(requestCmdJsonAuto);
+        }
+    }
+
+    @Override
+    public void initialize() {
+        // Set channel configuration parameters
+        channelConfigEhub = FerroampChannelConfiguration.getChannelConfigurationEhub();
+        channelConfigSsoS0 = FerroampChannelConfiguration.getChannelConfigurationSsoS0();
+        channelConfigSsoS1 = FerroampChannelConfiguration.getChannelConfigurationSsoS1();
+        channelConfigSsoS2 = FerroampChannelConfiguration.getChannelConfigurationSsoS2();
+        channelConfigSsoS3 = FerroampChannelConfiguration.getChannelConfigurationSsoS3();
+        channelConfigEso = FerroampChannelConfiguration.getChannelConfigurationEso();
+        channelConfigEsm = FerroampChannelConfiguration.getChannelConfigurationEsm();
+
+        ferroampConfig = getConfigAs(FerroampConfiguration.class);
```

Comment:
Don;t you need to check the configuration for sanity and set the thingstatus to CONFIGURATION_ERROR accordingly

Reply:
Ongoing 20240607